### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Katletos**
Wrote a script for the database. Tested with an online compiler and local machine.

image

### CODE REVIEW

File: database.sql [hunk 1]
```diff
@@ -0,0 +1,115 @@
+CREATE TABLE Brands (
+  brand_id bigserial,
+  name text NOT NULL,
+  PRIMARY KEY (brand_id)
+);
+
+CREATE TABLE Products (
+  product_id bigserial,
```
Comment (gregorzamza) [inline database.sql]: all PKs should be just id

File: database.sql [hunk 2]
```diff
@@ -0,0 +1,115 @@
+CREATE TABLE Brands (
+  brand_id bigserial,
+  name text NOT NULL,
```
Comment (gregorzamza) [inline database.sql]: missing unique constraints

File: database.sql [hunk 3]
```diff
@@ -0,0 +1,115 @@
+CREATE TABLE Brands (
```
Comment (gregorzamza) [inline database.sql]: all tables can start with small letter because they will be translated to it anyway

File: database.sql [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+);
+
+CREATE TYPE user_type AS ENUM ('admin', 'customer');
+
+CREATE TABLE Users (
+  user_id bigserial,
+  user_type user_type NOT NULL,
+  email text NOT NULL,
+  password text NOT NULL,
+  phone text NOT NULL,
```
Comment (gregorzamza) [inline database.sql]: it makes no sense to set this for admin for example

File: database.sql [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  url text NOT NULL,
+  file_type text NOT NULL,
+  file_name text NOT NULL,
+  PRIMARY KEY (image_id),
+  FOREIGN KEY (product_id) REFERENCES Products (product_id) ON DELETE CASCADE
+);
+
+CREATE TABLE Categories (
+  category_id bigserial,
+  parent_category_id bigint,
```
Comment (gregorzamza) [inline database.sql:87]: so it can be null?

File: database.sql [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  id bigserial,
+  name text NOT NULL UNIQUE,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE products (
+  id bigserial,
+  brand_id bigint NOT NULL,
+  category_id bigint NOT NULL,
+  price money NOT NULL,
```
Comment (gregorzamza) [inline database.sql:12]: I missed it on the diagram but we need at least name for the product

File: database.sql [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  id bigserial,
+  parent_category_id bigint,
+  name text NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (parent_category_id) REFERENCES categories (id)
+);
+
+CREATE TABLE sections (
+  id bigserial,
+  name text NOT NULL,
```
Comment (gregorzamza) [inline database.sql]: one more missing unique constraint

File: database.sql [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  file_type text NOT NULL,
+  file_name text NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (product_id) REFERENCES products (id) ON DELETE CASCADE
+);
+
+CREATE TABLE categories (
+  id bigserial,
+  parent_category_id bigint,
+  name text NOT NULL,
```
Comment (gregorzamza) [inline database.sql]: and one more

File: database.sql [hunk 9]
```diff
@@ -0,0 +1,116 @@
+CREATE TABLE brands (
+  id bigserial,
+  name text NOT NULL UNIQUE,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE products (
+  id bigserial,
+  brand_id bigint NOT NULL,
+  category_id bigint NOT NULL,
```
Comment (zhatvva) [inline database.sql:10]: I think you also should add fk for the category relation (same as for brands)

File: database.sql [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+);
+
+CREATE TABLE orders_items (
+  product_id bigint,
+  order_id bigint,
+  quantity bigint NOT NULL,
+  PRIMARY KEY (product_id, order_id),
+  FOREIGN KEY (product_id) REFERENCES products (id) ON DELETE RESTRICT,
+  FOREIGN KEY (order_id) REFERENCES orders (id) ON DELETE CASCADE
+);
```
Comment (zhatvva) [inline database.sql]: Should the price field also be added to the orders_items table (as i see on the other market platforms it is a common case then you can check a price of an every item you ordered)?

File: database.sql (hunk 9 above)
Comment (Katletos) [reply database.sql:10]: Ok, fixed

File: database.sql (hunk 10 above)
Comment (Katletos) [reply database.sql]: It makes sence

**Review by zhatvva — APPROVED** [review]

**Pull request merged**